db/migrate/20150404193023_participation_counter.rb

```diff
@@ -0,0 +1,36 @@
+class ParticipationCounter < ActiveRecord::Migration
+  def up
+    add_column :participations, :count, :int, null: false, default: 1
+    posts_count = Post.select("COUNT(posts.id)")
+                  .where("posts.id = participations.target_id")
+                  .where("posts.author_id = participations.author_id")
+                  .to_sql
+    likes_count = Like.select("COUNT(likes.id)")
+                  .where("likes.target_id = participations.target_id")
+                  .where("likes.author_id = participations.author_id")
+                  .to_sql
+    comments_count = Comment.select("COUNT(comments.id)")
+                     .where("comments.commentable_id = participations.target_id")
+                     .where("comments.author_id = participations.author_id")
+                     .to_sql
+    Participation.update_all("count = (#{posts_count}) + (#{likes_count}) + (#{comments_count})")
+
+    execute "update participations
+      set count = count + 1
+      where exists(
+        select 1
+        from poll_participations pp
+          inner join polls po
+            on po.id = pp.poll_id
+        where
+          pp.author_id = participations.author_id
+          and po.status_message_id = participations.target_id
+      )"
```

Comment:
Let's not mix both styles, either do it via AR or write a full query. Also I think this could still be one query.